src/pages/WatchVideo/WatchVideo.jsx

```diff
@@ -1,29 +1,78 @@
 import { Navbar, Sidebar, VideoCard } from "components";
-import { useVideosAndCategories } from "context";
-import React from "react";
-import { useParams, Link } from "react-router-dom";
+import { useVideosAndCategories, useVideosOperations, useAuth } from "context";
+import { useParams } from "react-router-dom";
 import styles from "./watchvideo.module.css";
 import { RiPlayListAddFill } from "react-icons/ri";
 import { AiFillLike, AiOutlineLike } from "react-icons/ai";
 import { BsStopwatch, BsStopwatchFill } from "react-icons/bs";
 import { getSimilarVideos } from "utils";
+import { useEffect } from "react";
+import axios from "axios";
+import { HISTORY_API } from "utils/APIEndPoints";
 
 export function WatchVideo() {
   const { videoID } = useParams();
   const {
     videosAndCategory: { videos },
   } = useVideosAndCategories();
 
+  const {
+    manageVideoLike,
+    manageWatchLater,
+    videosOperationsDispatcher,
+    videosOperations: { likedVideos, watchLaterVideos },
+  } = useVideosOperations();
+
+  const video = videos.find((video) => videoID === video.videoID);
   const {
     title,
     description,
     creator: { name: creatorName, profile: creatorProfile },
     views,
     category,
-  } = videos.find((video) => videoID === video.videoID);
+  } = video;
 
   const similarVideos = getSimilarVideos(category, videos, videoID);
 
+  const btnLikeHandler = () => {
+    manageVideoLike(video);
+  };
+
+  const btnWatchLaterHandler = () => {
+    manageWatchLater(video);
+  };
+  const {
+    auth: { encodedToken },
+  } = useAuth();
+  console.log(encodedToken);
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.post(
```

Comment:
Suggestion :- Try to extract this api call function

Reply:
The watch later API will run on page load of that particular video that's why I have put it there.